app/helpers/application_helper.rb

```diff
@@ -28,13 +28,37 @@ def iiif_drag_n_drop(manifest, width: '40px')
   ##
   # Renders a viewer for an object with understanding of the context. In the
   # context of spotlight/catalog render the configured viewer. In other contexts
-  # (feature page) render the default viewer
+  # (feature page) render the default viewer. Now passes through a "block" from
+  # SirTrevor, used for rendering viewers in specific ways (canvas index).
   # @param [SolrDocument] document
-  def render_viewer_in_context(document)
+  # @param [SirTrevorRails::Blocks::SolrDocumentsEmbedBlock] block
+  def render_viewer_in_context(document, block)
     if params[:controller] == 'spotlight/catalog'
-      render current_exhibit.required_viewer, document: document
+      render current_exhibit.required_viewer, document: document, block: block
     else
-      render current_exhibit.required_viewer.default_viewer_path, document: document
+      render current_exhibit.required_viewer.default_viewer_path, document: document, block: block
     end
   end
+
+  ##
+  #
+  # @param [SolrDocument] document
+  # @param [Integer] canvas_index
+  def custom_render_oembed_tag_async(document, canvas_index)
+    url = document.first(blacklight_config.show.oembed_field)
+
+    content_tag :div, '', data: { embed_url: blacklight_oembed_engine.embed_url(url: url, canvas_index: canvas_index) }
+  end
+
+  ##
+  # This method sends the message of which "canvas index" (zero-based) should be
+  # selected for a oembed viewer. For this it assumes that a
+  # SirTrevorRails::Blocks::SolrDocumentsEmbedBlock's first item has
+  # `iiif_canvas_id` and that `@id` conforms to the common DLSS `@id` format
+  # e.g. https://purl.stanford.edu/ab123cd4567/iiif/canvas/ab123cd4567_1
+  # @param [SirTrevorRails::Blocks::SolrDocumentsEmbedBlock] block
+  # @param [Integer]
+  def choose_canvas_index(sir_trevor_block)
```

Comment:
Ought we push this logic into `SirTrevorRails::Blocks::SolrDocumentsEmbedBlock`?

Reply:
Yeah I thought of that, but wasn't sure if it made sense. This method at the moment is kind of assuming our own canvas URI syntax. So it felt a little strange enforcing that upstream, but do you think a monkey patch in Exhibits would be better?

Reply:
Got it; nah, this is fine for now.